ehitchhiking-ui/src/app/pages/main-screen/yandex-map/yandex-map.component.ts

```diff
@@ -38,9 +44,46 @@ export class YandexMapComponent implements OnInit, OnChanges {
 		return Math.floor(Math.random() * (max - min)) + min;
 	}
 
+	placeMarkForPassenger(){
+    this.ymapsPromise
+      .then((maps) => {
+        const myGeoObject = new maps.GeoObject({
+          geometry: {
+            type: 'Point',
+            coordinates: [53.9, 27.5666]
+          },
+          properties: {
+            hintContent: 'Ну давай уже тащи'
+          }
+        }, {
+          preset: 'islands#yellowPersonIcon',
+          draggable: true
+        });
+        this.myMap.geoObjects
+          .add(myGeoObject);
+        this.myMark = myGeoObject;
+
+        myGeoObject.events.add('dragend', (event) => {
+          this.currentRoute.passengers[0] = {passanger: 'gleb',coordinates: this.myMark.geometry._coordinates};// TODO:0 is bad
+        });
+      });
+  }
+
 	ngOnChanges(changes: SimpleChanges) {
+		if (changes.triggers && changes.triggers.currentValue && this.isNeedCleanMap) {
+      this.yandexRoutesObjects.forEach((route) =>
```

Comment:
to my mind you shouldn't handle indexRouteToDisplay withing if (changes.triggers && changes.triggers.currentValue && this.isNeedCleanMap), it is a separate action

Reply:
done